2014_TPDS_basic_decomposition_based_algorithm/src/handle_segment.rs

```diff
@@ -16,9 +16,11 @@ pub struct Segment {
     pub nodes: Vec<NodeData>,
     pub begin_range: i32,
     pub end_range: i32,
-    pub deadline: f32, //TODO: use
+    pub deadline: f32,
     pub classification: Option<SegmentClassification>,
-    pub parallel_degree: i32,
+    pub execution_requirement: i32, // end_range - begin_range
+    pub parallel_degree: i32,       // number of nodes in the segment
+    pub volume: i32,                // execution_requirement * volume
```

Comment:
```suggestion
    pub volume: i32,                // execution_requirement * nodes.len()
```

Reply:
Correction omission. Thanks for pointing this out.